sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorClient/MetricFeedbackLiveTests.cs

```diff
@@ -0,0 +1,290 @@
+                options.StartTime = FeedbackSamplingStartTime;
+                options.EndTime = FeedbackSamplingEndTime;
+                options.FeedbackType = FeedbackType.Comment;
+
+                options.Filter.AddDimensionColumn("city", "Delhi");
+            }
+
+            var feedbackCount = 0;
+
+            await foreach (MetricFeedback feedback in client.GetAllFeedbackAsync(MetricId, options))
+            {
+                Assert.That(feedback, Is.Not.Null);
+                Assert.That(feedback.Id, Is.Not.Null.And.Not.Empty);
+                Assert.That(feedback.MetricId, Is.EqualTo(MetricId));
+                Assert.That(feedback.UserPrincipal, Is.Not.Null.And.Not.Empty);
+                Assert.That(feedback.CreatedTime, Is.Not.Null);
+
+                Assert.That(feedback.DimensionFilter, Is.Not.Null);
+                Assert.That(feedback.DimensionFilter.DimensionFilter, Is.Not.Null);
+
+                ValidateGroupKey(feedback.DimensionFilter.DimensionFilter);
+
+                if (populateOptionalMembers)
+                {
+                    Assert.That(feedback.CreatedTime, Is.GreaterThanOrEqualTo(FeedbackSamplingStartTime));
+                    Assert.That(feedback.CreatedTime, Is.LessThanOrEqualTo(FeedbackSamplingEndTime));
+                    Assert.That(feedback.Type, Is.EqualTo(FeedbackType.Comment));
+
+                    Dictionary<string, string> dimensionColumns = feedback.DimensionFilter.DimensionFilter.AsDictionary();
+
+                    Assert.That(dimensionColumns.ContainsKey("city"));
+                    Assert.That(dimensionColumns["city"], Is.EqualTo("Delhi"));
+                }
+
+                if (feedback.Type == FeedbackType.Anomaly)
+                {
+                    var anomalyFeedback = feedback as MetricAnomalyFeedback;
+
+                    Assert.That(anomalyFeedback, Is.Not.Null);
+                    Assert.That(anomalyFeedback.AnomalyValue, Is.Not.EqualTo(default(AnomalyFeedbackValue)));
+
+                    if (anomalyFeedback.AnomalyDetectionConfigurationId != null)
+                    {
+                        // TODO: Add snapshot validation (https://github.com/azure/azure-sdk-for-net/issues/15915).
+                    }
+                }
+                else if (feedback.Type == FeedbackType.ChangePoint)
+                {
+                    var changePointFeedback = feedback as MetricChangePointFeedback;
+
+                    Assert.That(changePointFeedback, Is.Not.Null);
+                    Assert.That(changePointFeedback.ChangePointValue, Is.Not.EqualTo(default(ChangePointValue)));
+                }
+                else if (feedback.Type == FeedbackType.Comment)
+                {
+                    var commentFeedback = feedback as MetricCommentFeedback;
+
+                    Assert.That(commentFeedback, Is.Not.Null);
+                    Assert.That(commentFeedback.Comment, Is.Not.Null.And.Not.Empty);
+                }
+                else
```

Comment:
do you think it is worth adding `else if (feedback.Type == Feedbacktype.Period)` and then if else is reached throw? just to catch a service error? (after our experience in FR, it might be worth adding)

Reply:
I have an assertion inside the `else` that causes the same effect:
```cs
Assert.That(feedback.Type, Is.EqualTo(FeedbackType.Period));
```